Assets/General/Scripts/ScreenText/DialoguePopUps.cs

```diff
@@ -0,0 +1,106 @@
+/*****************************************************************************
+// File Name :         DialoguePopUps.cs
+// Author :            Nick Rice
+//                     
+// Creation Date :     11/12/24
+//
+// Brief Description : This script handles the dialogue process, it's words and timing
+*****************************************************************************/
+using System;
+using System.Collections;
+using TMPro;
+using Unity.VisualScripting;
+using UnityEngine;
+
+/// <summary>
+/// This class handles the dialogue process, it's words and timing
+/// </summary>
+public class DialoguePopUps : MonoBehaviour
+{
+    [Tooltip("The UI element that will actually display the text")]
+    [SerializeField]
+    TextMeshProUGUI _textContainer;
+
+    // Temp variable used for testing
+    [Tooltip("The data used in the UI element")]
+    [SerializeField]
+    ScriptableDialogueUI _uIData;
+
+    [Tooltip("The collider the player has to walk into")]
+    GameObject _walkTutorialObject;
+
+    [Tooltip("The collider the player has to shoot")]
+    GameObject _shootTutorialObject;
+
+    [Tooltip("The pointer for which ui data is currently being used")]
+    private int _dataPointer = 0;
+    
+    /// <summary>
+    /// Temporary; used for testing
+    /// </summary>
+    private void Start()
+    {
+        GameStateManager.Instance.GetOnNewDialogueChain()?.Invoke(_uIData);
+    }
+
+    /// <summary>
+    /// The pass through function for actually displaying the dialogue
+    /// Because events do not like coroutines
+    /// </summary>
+    private void FunctionForDisplayingText(ScriptableDialogueUI dialogueUI)
+    {
+        StartCoroutine(DisplayingTheText(dialogueUI));
+    }
+
+    /// <summary>
+    /// This takes the text, makes it invisible, then slowly makes it visible by x characters a second
+    /// </summary>
+    /// <returns></returns>
+    private IEnumerator DisplayingTheText(ScriptableDialogueUI moreDialogue)
```

Comment:
Maybe rename this to "DisplayTest"

Reply:
No; it's not a test; only the start function is

Reply:
I meant to say `DisplayText`, that was a typo on my part